### PR TITLE
Apr 1941 add clarification and outcome

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Roatp/RoatpFinancialControllerTests.cs [hunk 1]
```diff
@@ -27,22 +30,40 @@ public class RoatpFinancialControllerTests
...
+            _financialReviewDetails = new FinancialReviewDetails();
+            MockHttpContextAccessor = SetupMockedHttpContextAccessor();
+
             _controller = new RoatpFinancialController(_roatpOrganisationApiClient.Object,
                 _applicationApplyApiClient.Object,
                 _qnaApiClient.Object,
-                Mock.Of<IHttpContextAccessor>());
+                MockHttpContextAccessor.Object)
+            {
+                ControllerContext = new ControllerContext {HttpContext = new DefaultHttpContext()}
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Controllers/Roatp/RoatpFinancialControllerTests.cs:53]: There's a helper method in `SFA.DAS.AdminService.Common.Testing.MockedObjects` for this

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Roatp/RoatpFinancialControllerTests.cs [hunk 2]
```diff
@@ -27,22 +30,40 @@ public class RoatpFinancialControllerTests
...
+            var user = MockedUser.Setup();
 
-            _controller.ControllerContext = new ControllerContext();
-            _controller.ControllerContext.HttpContext = new DefaultHttpContext();
+            var mockHttpContextAccessor = new Mock<IHttpContextAccessor>();
+            var context = new DefaultHttpContext { User = user };
 
+            mockHttpContextAccessor.Setup(_ => _.HttpContext).Returns(context);
+            return mockHttpContextAccessor;
         }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/Controllers/Roatp/RoatpFinancialControllerTests.cs:67]: This looks like a code smell. Controllers don't need a HttpContextAccessor as it's part of the ControllerContext

File: src/SFA.DAS.AdminService.Web.Tests/SFA.DAS.AdminService.Web.Tests.csproj [hunk 3]
```diff
@@ -15,6 +15,7 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\SFA.DAS.AdminService.Application.Interfaces\SFA.DAS.AdminService.Application.Interfaces.csproj" />
+    <ProjectReference Include="..\SFA.DAS.AdminService.Common.Testing\SFA.DAS.AdminService.Common.Testing.csproj" />
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web.Tests/SFA.DAS.AdminService.Web.Tests.csproj:18]: Technically should be using the NuGet version of this but fine for now

File: src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpFinancialController.cs [hunk 4]
```diff
@@ -320,6 +378,26 @@ private async Task<List<Section>> GetFinancialSections(RoatpApplicationResponse
             return financialSections;
         }
 
+
+        private RoatpFinancialClarificationViewModel ConvertFinancialApplicationToFinancialClarificationViewModel(RoatpFinancialApplicationViewModel vm)
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpFinancialController.cs:382]: Can you mark it static

File: src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Financial/RoatpFinancialApplicationViewModel.cs [hunk 5]
```diff
@@ -10,7 +10,7 @@ namespace SFA.DAS.AdminService.Web.ViewModels.Apply.Financial
 {
     public class RoatpFinancialApplicationViewModel : OrganisationDetailsViewModel
     {
-        public List<Section> Sections { get; }
+        public List<Section> Sections { get; set; }
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Financial/RoatpFinancialApplicationViewModel.cs:13]: I couldn't see why the set is required here. Is there a need to replace/initialise the entire list?

**Review by martin-hornsby — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AdminService.Web.Tests/Controllers/Roatp/RoatpFinancialControllerTests.cs (hunk 2 above)
Comment (martin-hornsby) [reply src/SFA.DAS.AdminService.Web.Tests/Controllers/Roatp/RoatpFinancialControllerTests.cs:67]: On second look it looks like the controller had this property previously 🤢

File: src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpFinancialController.cs [hunk 6]
```diff
@@ -18,6 +18,11 @@
 using System.IO.Compression;
 using System.Linq;
 using System.Threading.Tasks;
+using OfficeOpenXml.VBA;
```
Comment (martin-hornsby) [inline src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpFinancialController.cs:21]: Can you remove this

**Review by martin-hornsby — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Financial/RoatpFinancialApplicationViewModel.cs (hunk 5 above)
Comment (MarkFCain) [reply src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Financial/RoatpFinancialApplicationViewModel.cs:13]: yes, I need to map them back in the ConvertFinancialApplicationToFinancialClarificationViewModel

Comment (MarkFCain) [reply src/SFA.DAS.AdminService.Web/ViewModels/Roatp/Financial/RoatpFinancialApplicationViewModel.cs:13]: which inherts from RoatpFinancialApplicationViewModel

**Review by martin-hornsby — DISMISSED** [review]

**Review by martin-hornsby — APPROVED** [review]

**Review by martin-hornsby — APPROVED** [review]

**Pull request merged**